docker/mongodb-kubernetes-tests/kubetester/mongodb_common.py

```diff
@@ -11,20 +11,26 @@
 
 class MongoDBCommon:
     @TRACER.start_as_current_span("wait_for")
-    def wait_for(self, fn, timeout=None, should_raise=True):
+    def wait_for(self, fn, timeout=None, should_raise=True, persist_for=1):
```

Comment:
Can we separate "waiting for something" from "something staying in the same state for a period of time" please? It will keep the testing framework simple and understanding of tests easier: no need to dig for the meaning of `persist_for` argument.

Here is an example of how it is done in Gomega:
* https://onsi.github.io/gomega/#eventually - our `wait_for`
* https://onsi.github.io/gomega/#consistently - what you are trying to incorporate into `wait_for`